sdk/maps/azure-maps-geolocation/src/test/java/com/azure/maps/geolocation/GeolocationClientTestBase.java

```diff
@@ -3,95 +3,116 @@
 
-    private final String endpoint = Configuration.getGlobalConfiguration().get("API-LEARN_ENDPOINT");
-    Duration durationTestMode;
     static InterceptorManager interceptorManagerTestBase;
 
+    Duration durationTestMode;
+
     @Override
     protected void beforeTest() {
         if (interceptorManager.isPlaybackMode()) {
             durationTestMode = Duration.ofMillis(1);
         } else {
             durationTestMode = TestUtils.DEFAULT_POLL_INTERVAL;
         }
+
         interceptorManagerTestBase = interceptorManager;
     }
 
-    GeolocationClientBuilder getGeoLocationAsyncClientBuilder(HttpClient httpClient, GeolocationServiceVersion serviceVersion) {
+    GeolocationClientBuilder getGeoLocationAsyncClientBuilder(HttpClient httpClient,
+                                                              GeolocationServiceVersion serviceVersion) {
         GeolocationClientBuilder builder = new GeolocationClientBuilder()
+            .pipeline(getHttpPipeline(httpClient))
             .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
             .serviceVersion(serviceVersion);
-        String endpoint = getEndpoint();
-        if (getEndpoint() != null) {
-            builder.endpoint(endpoint);
-        }
-        if (getTestMode() == TestMode.RECORD) {
-            builder.addPolicy(interceptorManager.getRecordPolicy());
-        }
-        if (getTestMode() == TestMode.PLAYBACK) {
-            builder.credential(new AzureKeyCredential(FAKE_API_KEY)).httpClient(interceptorManager.getPlaybackClient());
-        } else {
-            builder.credential((new AzureKeyCredential(
-                Configuration.getGlobalConfiguration().get("SUBSCRIPTION_KEY"))));
+
+        if (interceptorManager.isPlaybackMode()){
+            builder.endpoint("https://localhost:8080");
         }
+
         return builder;
     }
 
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode()) {
+            interceptorManager.addSanitizers(
+                Collections.singletonList(
+                    new TestProxySanitizer("subscription-key", ".+", "REDACTED", TestProxySanitizerType.HEADER)));
         }
 
-        final List<HttpPipelinePolicy> policies = new ArrayList<>();
-        if (credential != null) {
-            policies.add(new BearerTokenAuthenticationPolicy(credential, endpoint.replaceFirst("/$", "") + "/.default"));
+        if (interceptorManager.isPlaybackMode()) {
+            List<TestProxyRequestMatcher> customMatchers = new ArrayList<>();
+
+            customMatchers.add(new BodilessMatcher());
```

Comment:
do we need the bodiless matcher?

Reply:
I can try running tests without it. I found I needed it for Key Vault so I played it safe here.